docs/integrations/ml-dist.rst

```diff
@@ -0,0 +1,512 @@
+training with the ``EarlyStopping`` callback.
+
+.. code-block:: python
+
+    import tensorflow as f
+    import matplotlib.pyplot as plt
+    from tensorflow.python.keras.callbacks import EarlyStopping
+    from tensorflow.keras import layers
+
+
+    mlp_model = tf.keras.Sequential()
+
+    mlp_model.add(layers.Dense(200, activation="relu", input_shape=(input_len,)))
+    mlp_model.add(layers.Dense(40, activation="relu"))
+    mlp_model.add(layers.Dense(1))
+
+    mlp_model.compile(optimizer="adam",
+                      loss=tf.keras.losses.MeanSquaredError(),
+                      metrics=["accuracy"])
+
+    es = EarlyStopping(monitor="loss", mode="auto", verbose=1, patience=50)
+    hist = mlp_model.fit(x_train, y_train, epochs=500, callbacks=[es])
+
+    plt.plot(hist.history["loss"])
+    plt.show()
+
+.. figure:: mlp_model_train_loss.png
+   :align: left
+
+Now that we have our MLP model, we can start preparing the test data to verify
+the accuracy of the model.
+
+.. code-block:: python
+
+    test_pairs = []
+
+    test_saved_step = 0
+    for i in df_test.index:
+        row = test_saved_step
+        if row + timestep + output_timestep < len(df_test.index):
+            data_window = norm_test_data_cols[row:row + timestep]
+            data_vector = data_window.flatten()
+            state = add_labels(row + timestep, row + timestep + output_timestep, df_test)
+            test_pairs.append((data_vector, state))
+            test_saved_step = row + timestep
+        else:
+            break
+
+    x_test = np.zeros((len(test_pairs), input_len))
+    y_test = np.zeros((len(test_pairs),))
+
+    for index in range(0, len(test_pairs)):
+        (x, y) = test_pairs[index]
+        x_test[index, 0:input_len] = x
+        y_test[index] = y
+
+The data is prepared in the same way as the training data was prepared, and now we can
+predict the test data and look at the accuracy.
+
+.. code-block:: python
+
+    predictions = mlp_model.predict(x_test)
+    predictions = predictions.flatten()
+
+    _, train_acc = mlp_model.evaluate(x_train, y_train, verbose=0)
+    _, test_acc = mlp_model.evaluate(x_test, y_test, verbose=0)
+
+    print("train_acc: ", train_acc)
+    print("test_acc: ", test_acc)
+
+    [1] train_acc: 0.9827990531921387
+    [2] test_acc:  0.8741965293884277
+
+The accuracy will vary slightly between the different executions. The train accuracy
+is nearly perfect, while the test accuracy is at 87%. These percentages apply when the
+training input is 60 minutes and the prediction output would be for the next 720 minutes
+(12 hours). Depending on how many epochs one uses and the size of the timesteps the results
+vary as well.
+
+Saving A Model To S3
+--------------------
+
+The next step is to save the model to a AWS S3 Bucket so we can access the model from
+potentially another application. Since the data we want to predict will not be normalised
```

Comment:
```suggestion
The next step is to save the model to a AWS S3 Bucket so the model can be accessed from
other applications. Since the data we want to predict will not be normalised
```